### PR TITLE
Use vendored OpenSSL with reqwest [ci full]

### PR DESCRIPTION
**Pull request by eoger**
This is an alternative fix to https://github.com/mozilla/application-services/issues/1432: since we can't get rid of OpenSSL entirely yet, we can still remove the requirement to build it ourselves and use [the vendored version of it](https://github.com/alexcrichton/openssl-src-rs/).

### CODE REVIEW

**Comment by eoger** [comment]
Ready for review, this PR only affect the last `Desktop (Linux)` line from the table at https://github.com/mozilla/application-services/issues/1432#issue-470478877.

File: libs/build-all-android.sh [hunk 1]
```diff
@@ -37,18 +36,6 @@ for i in "${!TARGET_ARCHS[@]}"; do
   fi
 done
 
-echo "# Building openssl"
```
Comment (vladikoff) [inline libs/build-all-android.sh]: 💟

**Comment by rfk** [comment]
Hrm, taskcluster failed to compile "logins" with some sort of include-file-not-found error. Going to see if I can re-trigger it and whether it's spurious...

**Comment by eoger** [comment]
> Hrm, taskcluster failed to compile "logins" with some sort of include-file-not-found error. Going to see if I can re-trigger it and whether it's spurious...

Thanks, looks like it's passing now.

**Comment by eoger** [comment]
> r+, but could you please also update the dependency script to indicate that the `openssl-src` crate pulls in a hidden dependency on `ext-openssl`, for completeness?

I think it's actually better to leave it on `openssl-sys` as it works even if you are using your own OpenSSL like we used to do.

**Pull request merged**